packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/wc_sessionRequest.ts

```diff
@@ -200,7 +201,14 @@ class WCSessionRequestHandler implements RpcRequestHandler<WCSessionProposal> {
     const { proposer, requiredNamespaces, optionalNamespaces } = params
 
     const normalizedRequired = normalizeNamespaces(requiredNamespaces)
-    const normalizedOptional = normalizeNamespaces(optionalNamespaces)
+    const normalizedOptional = normalizeNamespaces({
+      ...optionalNamespaces,
+      // add optional namespaces for non-evm chains support
+      // since core web integrated wagmi and it only supports EVM for now,
+      // it throws an error when we add these non-EVM namespaces in the dapp.
+      // This is a temporary fix until core web supports these namespaces
+      ...NONEVM_OPTIONAL_NAMESPACES
```

Comment:
this should only be added for core apps

Reply:
fixed it to add NONEVM_OPTIONAL_NAMESPACES only when isCoreDomain is true